### ISSUE TITLE
Non mandatory plugins intermittently stop working

### ISSUE THREAD

**Issue by msvticket** [jenkins-x/jx#7922]
Now and then non mandatory plugins of jx stop working:

```
$ jx updatebot
Error: unknown command "updatebot" for "jx"
Run 'jx --help' for usage.
```

I've seen the same with jx-registry, but not with mandatory plugins like jx-pipeline. The plugins are still executable from ~/.jx3/plugins/bin:

```
$ ~/.jx3/plugins/bin/jx-updatebot-0.1.1
commands for creating Pull Requests on repositories when versions change
---
```

I first say the behaviour in a container where jx registry is repeatedly called. For the first call the plugin was downloaded and used. It worked for a couple of more times before it stopped working with the unknown command error.

When it has stopped working it tend to continue to fail for the rest of the day, but start working again the next day.

For reference:
```
jx version
version: 3.2.185

```

**Comment by msvticket** [comment]
After digging in the code I realised github is called for every call to a plugin not listen in the plugins package. This made me believe the problem was githubs API rate limit. And after adding some better error handling in I found this to be the case:

ERROR: failed to load plugin jx-updatebot: can't find version: {"message":"API rate limit exceeded for xxx.xxx.xxx.xxx. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)","documentation_url":"https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting"}
